### PR TITLE
scsi: Support updating firmware on UFS devices

### PR DESCRIPTION
**Pull request by hughsie**
Change-Id: I5573b1890135d1fd3b2f0cb8f95f74cbb3e21abe

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/scsi/README.md [hunk 1]
```diff
@@ -2,8 +2,19 @@
 
 ## Introduction
 
-This plugin adds support for SCSI storage hardware. Devices are only enumerated
-and are not updatable.
+This plugin adds support for SCSI storage hardware. Most SCSI devices are are enumerated and UFS
```
Comment (daniil-lunev) [inline plugins/scsi/README.md:5]: s/are are/are/

File: plugins/scsi/fu-scsi-device.c [hunk 2]
```diff
@@ -46,9 +70,36 @@ fu_scsi_device_probe(FuDevice *device, GError **error)
 				    "no assigned vendor");
 		return FALSE;
 	}
+
 	vendor_id = g_strdup_printf("SCSI:%s", fu_device_get_vendor(device));
 	fu_device_add_vendor_id(device, vendor_id);
 
+	/* the controller could really on any bus.. */
```
Comment (daniil-lunev) [inline plugins/scsi/fu-scsi-device.c:77]: could really be*

 May be word it as a TODO to add better search for ufshci?

**Review by daniil-lunev — APPROVED** [review]
LGTM modulo small comment nits

One Q, not related to this CL - I opbserved update with fwupdtool doesn't actually update the history? i.e. I see need-update flag in verbose log of install-blob, but then it is not reflected in the get-devices, is that expected?

File: plugins/scsi/README.md (hunk 1 above)
Comment (hughsie) [reply plugins/scsi/README.md:5]: Fixed, thanks.

File: plugins/scsi/fu-scsi-device.c (hunk 2 above)
Comment (hughsie) [reply plugins/scsi/fu-scsi-device.c:77]: Agree, fixed.

**Comment by hughsie** [comment]
> fwupdtool doesn't actually update the history

Ya expected, you need to use `install` rather than `install-blob` if you want to use the history functionality.

**Pull request merged**